ubirch/crypto.go

```diff
@@ -321,29 +321,32 @@ func (c *CryptoContext) GetPublicKey(name string) ([]byte, error) {
 }
 
 // getDecodedPrivateKey gets the decoded private key for the given name.
-func (c *CryptoContext) getDecodedPrivateKey(name string) (*ecdsa.PrivateKey, error) {
-	id, err := c.GetUUID(name)
-	if err != nil {
-		return nil, err
-	}
+func (c *CryptoContext) getDecodedPrivateKey(id uuid.UUID) (*ecdsa.PrivateKey, error) {
 	//check for invalid keystore
 	if c.Keystore == nil { //check for 'direct' nil
 		return nil, fmt.Errorf("can't get private key: keystore is nil")
 	} else if reflect.ValueOf(c.Keystore).IsNil() { //check for pointer which is nil
 		return nil, fmt.Errorf("can't get private key: keystore pointer is nil, pointer type is %T", c.Keystore)
 	}
+
+	// get encoded private key from keystore
 	privKey, err := c.Keystore.GetKey(privKeyEntryTitle(id))
 	if err != nil {
 		return nil, err
 	}
 
 	// decode the key
-	return decodePrivateKey(privKey) //todo, check if this is necessary
+	return decodePrivateKey(privKey)
 }
 
 // PrivateKeyExists Checks if a private key entry for the given name exists in the keystore.
 func (c *CryptoContext) PrivateKeyExists(name string) bool {
-	_, err := c.getDecodedPrivateKey(name)
+	id, err := c.GetUUID(name)
```

Comment:
What is the reason, that you still ask for the Key with `name` and not `uid`? Would it make sense to simplify this, the same way, you simplified `getDecodedPrivateKey()`?

Reply:
for now i only refactored the private method `getDecodedPrivateKey` without changing the interface to the CryptoContext. 

`PrivateKeyExists(name string)` is a public method, so if we change the signature of `PrivateKeyExists` to expect a UUID, we would move the call of `c.GetUUID(name)` outside the CryptoContext. I think, it is better to leave it inside for now.

Reply:
I agree